src/kbmod/search/raw_image_eigen.cpp

```diff
@@ -0,0 +1,767 @@
+#include "raw_image_eigen.h"
+
+
+namespace py = pybind11;
+
+
+namespace search {
+  RawImageEigen::RawImageEigen()
+    : width(0),
+      height(0),
+      obstime(-1.0),
+      image() {}
+
+
+  RawImageEigen::RawImageEigen(Image& img, double obs_time)  {
+    image = std::move(img);
+    height = image.rows();
+    width = image.cols();
+    obstime = obs_time;
+  }
+
+
+  RawImageEigen::RawImageEigen(unsigned w, unsigned h, float value, double obs_time)
+    : height(h),
+      width(w),
+      obstime(obs_time) {
+    if (value != 0)
+      image = Image::Constant(height, width, value);
+    else
+      image = Image::Zero(height, width);
+  }
+
+
+  // Copy constructor
+  RawImageEigen::RawImageEigen(const RawImageEigen& old) {
+    width = old.get_width();
+    height = old.get_height();
+    image = old.get_array();
+    obstime = old.get_obstime();
+  }
+
+
+  // Move constructor
+  RawImageEigen::RawImageEigen(RawImageEigen&& source)
+    : width(source.width),
+      height(source.height),
+      obstime(source.obstime),
+      image(std::move(source.image)) {}
+
+
+  // Copy assignment
+  RawImageEigen& RawImageEigen::operator=(const RawImageEigen& source) {
+    width = source.width;
+    height = source.height;
+    image = source.image;
+    obstime = source.obstime;
+    return *this;
+  }
+
+
+  // Move assignment
+  RawImageEigen& RawImageEigen::operator=(RawImageEigen&& source) {
+    if (this != &source) {
+      width = source.width;
+      height = source.height;
+      image = std::move(source.image);
+      obstime = source.obstime;
+    }
+    return *this;
+  }
+
+
+  bool RawImageEigen::isclose(const RawImageEigen& img_b, float atol) const {
+    return image.isApprox(img_b.image, atol);
```

Comment:
This is much nicer than the double for loops.

Reply:
Unfortunately, because of the way we do things with `NO_DATA `there do not seem to be many opportunities to forward the work to Eigen. I'm leaving this as a note to myself to implement $L_\infty$ norm.